### PR TITLE
Hydrator Transformation Plugin Collection PR#1

### PR DESCRIPTION
**Pull request by nitinmotgi**
This PR is complete and ready for review.
- [x] Renamed package to co.cask.hydrator.plugins.*
- [x] Move groupId to 'co.cask.hydrator'
- [x] Move artifactId to 'hydrator-plugins' for parent
- [x] Added transform-plugins 
- [x] Added CSVFormatter required in transform-plugins
- [x] Add CSVParser

Following will be added in few upcoming PRs. 

PR #1
- [ ] Add Hasher
- [ ] Add Masker

PR #2
- [ ] Add Compressor
- [ ] Add Decompressor

PR #3
- [ ] Add JSON Parser
- [ ] Add JSON Formatter

PR #4
- [ ] Add Encoder
- [ ] Add Decoder

PR #5
- [ ] Add Identity
- [ ] Add StreamFormatter

PR #6
- [ ] Add Tokenizer

### CODE REVIEW

**Comment by awholegunch** [comment]
Should there not be a README.rst for the transform-plugins?

**Comment by awholegunch** [comment]
Changes to reference this could be added to https://github.com/caskdata/hydrator-plugins/blob/develop/README.rst...

File: transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVFormatter.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+ *   <li>EXCEL</li>
+ *   <li>RFC4180</li>
+ *   <li>MYSQL and</li>
+ *   <li>TDF</li>
+ * </ul>
+ * </p>
+ */
+@Plugin(type = "transform")
+@Name("CSVFormatter")
+@Description("Formats a Structure Record to CSV")
```
Comment (sreevatsanraman) [inline transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVFormatter.java]: Structured

File: transform-plugins/src/test/java/co/cask/hydrator/plugin/CSVFormatterTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    emitter.clear();
+    transform.transform(StructuredRecord.builder(INPUT1)
+                          .set("a", "6")
+                          .set("b", "7")
+                          .set("c", "8")
+                          .set("d", "9")
+                          .set("e", "10").build(), emitter);
+    Assert.assertEquals("6|7|8|9|10\r\n", emitter.getEmitted().get(0).get("body"));
+  }
+}
```
Comment (sreevatsanraman) [inline transform-plugins/src/test/java/co/cask/hydrator/plugin/CSVFormatterTest.java]: new line.

**Comment by sreevatsanraman** [comment]
LGTM

**Pull request merged**